program/rust/src/processor/upd_price.rs

```diff
@@ -126,31 +163,79 @@ pub fn upd_price(
         }
     }
 
-    #[cfg(test)]
-    // Sma feature disabled in production for now
-    {
-        use crate::accounts::PythAccount;
-        let account_len = price_account.try_data_len()?;
-        if aggregate_updated
-            && account_len == crate::time_machine_types::PriceAccountWrapper::MINIMUM_SIZE
-        {
-            let mut price_account = load_checked::<crate::time_machine_types::PriceAccountWrapper>(
-                price_account,
-                cmd_args.header.version,
+    let mut price_data = load_checked::<PriceAccount>(price_account, cmd_args.header.version)?;
+    // We want to send a message every time the aggregate price updates. However, during the migration,
+    // not every publisher will necessarily provide the accumulator accounts. The message_sent_ flag
+    // ensures that after every aggregate update, the next publisher who provides the accumulator accounts
+    // will send the message.
+    if aggregate_updated {
+        price_data.message_sent_ = 0;
+    }
+
+    if let Some(accumulator_accounts) = maybe_accumulator_accounts {
```

Comment:
this logic got relatively tricky and it could definitely use a unit test. Unfortunately it's quite difficult to add such a test without the message buffer program. I'll try to put one in the integration test repo.